pennylane/qchem/factorization.py

```diff
@@ -274,44 +276,128 @@ def basis_rotation(one_electron, two_electron, tol_factor=1.0e-5):
         eigenvector of the corresponding :math:`T` or :math:`L^{(r)}` matrix.
     """
 
-    t_matrix = one_electron - 0.5 * np.einsum("illj", two_electron)
-    two_electron = np.swapaxes(two_electron, 1, 3)
+    num_orbitals = one_electron.shape[0] * 2
+    one_body_tensor, chemist_two_body_tensor = _chemist_transform(one_electron, two_electron)
+    chemist_one_body_tensor = np.kron(one_body_tensor, np.eye(2))  # account for spin
+    t_eigvals, t_eigvecs = np.linalg.eigh(chemist_one_body_tensor)
```

Comment:
Is `chemist_one_body_tensor` hermitian complex or real symmetric? I'm just curious why we use `eigh` sometimes and `eig` other times.

Reply:
Yes, I think they should be real symmetric. Updated to `eigh` for consistency.